server/src/test/java/org/elasticsearch/action/admin/indices/alias/GetAliasesResponseTests.java

```diff
@@ -0,0 +1,138 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.indices.alias;
+
+import org.elasticsearch.action.admin.indices.alias.get.GetAliasesResponse;
+import org.elasticsearch.cluster.metadata.AliasMetaData;
+import org.elasticsearch.cluster.metadata.AliasMetaData.Builder;
+import org.elasticsearch.common.collect.ImmutableOpenMap;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.test.AbstractStreamableXContentTestCase;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+
+public class GetAliasesResponseTests extends AbstractStreamableXContentTestCase<GetAliasesResponse> {
+
+    @Override
+    protected GetAliasesResponse doParseInstance(XContentParser parser) {
+        try {
+            return GetAliasesResponse.fromXContent(parser);
+        } catch (IOException e) {
+            fail(e.getMessage());
+            return null;
+        }
+    }
+
+    @Override
+    protected GetAliasesResponse createTestInstance() {
+        return createTestItem();
+    }
+
+    @Override
+    protected GetAliasesResponse createBlankInstance() {
+        return new GetAliasesResponse(null);
+    }
+
+    @Override
+    protected GetAliasesResponse mutateInstance(GetAliasesResponse response) {
+        return new GetAliasesResponse(mutateAliases(response.getAliases()));
+    }
+
+    private static ImmutableOpenMap<String, List<AliasMetaData>> mutateAliases(ImmutableOpenMap<String, List<AliasMetaData>> aliases) {
+        if (aliases.isEmpty()) {
+            return addIndices(1, 3).build();
+        }
+
+        if (randomBoolean()) {
+            ImmutableOpenMap.Builder<String, List<AliasMetaData>> builder = ImmutableOpenMap.builder(aliases);
+            ImmutableOpenMap<String, List<AliasMetaData>> list = addIndices(1, 1).build();
+            list.forEach(e -> builder.put(e.key, e.value));
+            return builder.build();
+        }
+
+        Set<String> indices = new HashSet<>();
+        Iterator<String> keys = aliases.keysIt();
+        while (keys.hasNext()) {
+            indices.add(keys.next());
+        }
+
+        List<String> indicesToBeModified = randomSubsetOf(randomIntBetween(1, indices.size()), indices);
+        ImmutableOpenMap.Builder<String, List<AliasMetaData>> builder = ImmutableOpenMap.builder();
+
+        for (String index : indices) {
+            List<AliasMetaData> list = new ArrayList<>(aliases.get(index));
+            if (indicesToBeModified.contains(index)) {
+                list.add(createAliasMetaData());
+            }
+            builder.put(index, list);
+        }
+        return builder.build();
+    }
+
+    @Override
+    protected Predicate<String> getRandomFieldsExcludeFilter() {
+        return p -> p.equals("") // do not add new indices
```

Comment:
when is the field name empty? at the root of the object?

Reply:
For the top elements. Basically if it is not excluded, the elements added at this level are interpreted as a new index and as the index name is a free text field name, I  cannot ignore it in the `fromXContent`.

Reply:
makes sense. maybe expand a bit the comment?